python/lsst/ts/hexrotcomm/command_telemetry_client.py

```diff
@@ -352,8 +359,12 @@ async def read_telemetry_and_config(self):
                     await tcpip.read_into(self.telemetry_reader, self.config)
                     try:
                         self.config_callback(self)
-                    except Exception:
+                        if not self.configured_task.done():
+                            self.configured_task.set_result(None)
```

Comment:
Do you need this if you use the `make_done_future()`?

Reply:
Yes. You should *always* guard any `future.set_result` or `future.set_exception` with `if not future.done()` because it is an error to call either of those two methods on a done future. (Unlike `future.cancel()`, which is always safe).

In fact this code block would be a no-op if I initialized `configured_task` to a done task and never replaced it with a new Future.